src/DataReader/parse_queries.rs

```diff
@@ -172,6 +175,53 @@ fn build_expression_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpress
     }
 }
 
+fn build_state_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpression {
+    let mut inner_pair = pair.clone().into_inner();
+    let locPair = inner_pair.next().unwrap();
+
+    let mut loc_names: Vec<Box<QueryExpression>> = Vec::new();
+    for loc_name in locPair.as_str().split(',') {
+        loc_names.push(Box::new(QueryExpression::LocName(
+            loc_name.trim().to_string(),
+        )));
+    }
+
+    let clockPair = inner_pair.next().unwrap();
+
+    // In the following line of code, we build a BoolExprssion based on the clock constraints defined for the given location.
+    // To make BoolExprssion we use the InvariantParser parser instead.
+    // Becuase clocks is defined as c1&&c2... in the InvariantParser we replace ',' to match the format e.g., e.g., "x>0,y<5" => "x>0&&y<5"
+    let invariantVersion: Option<Box<BoolExpression>> = if clockPair.as_str().trim() != "" {
+        let clock_string = clockPair.as_str().trim().to_string().replace(',', "&&");
+        let invariant_version = parse(&clock_string).expect("");
+        Some(Box::new(invariant_version))
+    } else {
+        None
+    };
+
+    match pair.as_rule() {
+        Rule::state => QueryExpression::State(loc_names, invariantVersion),
+        err => panic!("Unable to match: {:?} as rule loc or clocks", err),
+    }
+}
+
+fn build_reachability_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpression {
+    let mut inner_pair = pair.into_inner();
+    let automata_pair = inner_pair.next().unwrap();
+    let s_state_pair = inner_pair.next().unwrap();
+    let e_state_pair = inner_pair.next().unwrap();
```

Comment:
Just spell it out instead of using a single letter do denote what it is

Reply:
Done